packages/SwingSet/src/vats/network/network.js

```diff
@@ -235,18 +236,18 @@ export function makeNetworkProtocol(protocolHandler) {
    */
   const bind = async localAddr => {
     // Check if we are underspecified (ends in slash)
-    if (localAddr.endsWith(ENDPOINT_SEPARATOR)) {
-      for (;;) {
-        // eslint-disable-next-line no-await-in-loop
-        const portID = await E(protocolHandler).generatePortID(
-          localAddr,
-          protocolHandler,
-        );
-        const newAddr = `${localAddr}${portID}`;
-        if (!boundPorts.has(newAddr)) {
-          localAddr = newAddr;
-          break;
-        }
+    const untilSpecified = asyncGenerate(() => ({
+      done: !localAddr.endsWith(ENDPOINT_SEPARATOR),
+      value: null,
+    }));
+    for await (const _ of untilSpecified) {
+      const portID = await E(protocolHandler).generatePortID(
+        localAddr,
+        protocolHandler,
+      );
+      const newAddr = `${localAddr}${portID}`;
+      if (!boundPorts.has(newAddr)) {
+        localAddr = newAddr;
```

Comment:
To figure out that this is correctness preserving, I see that `generatePortID` returns a non-empty string that does not end in slash. Assuming that's correct, no change suggested.

Reply:
I reworked this logic to be more closely derived from the old `if/while` construct.